src/engine.h

```diff
@@ -97,17 +97,14 @@ class Engine {
   virtual Result DoPatchParameterVertices(
       const PatchParameterVerticesCommand* cmd) = 0;
 
-  // Execute the probe command
-  virtual Result DoProbe(const ProbeCommand* cmd) = 0;
-
-  // Execute the probe ssbo command
-  virtual Result DoProbeSSBO(const ProbeSSBOCommand* cmd) = 0;
-
   // Execute the buffer command
   virtual Result DoBuffer(const BufferCommand* cmd) = 0;
 
-  // Execute the tolerance command
-  virtual Result DoTolerance(const ToleranceCommand* cmd) = 0;
+  // Run all queued commands and copy result data to the host.
+  virtual Result DoProcessCommands(uint32_t* stride,
```

Comment:
This is yucky but need to figure out a better way to do it. I need to get all the resources to handle the probe, which in this case is the size, stride and the bits to compare.

Reply:
The solution to this maybe to have VkScript use a FormatBuffer we haven't created yet. That format buffer will then have the bits filled in by the engine. It knows it's stride and size so you should then have all the things.

For VkScript we just can only have one output buffer, with AmberScript the EXPECT command tells the buffer to expect against.

This would be a followup change to implement FormatBuffer and use.

Reply:
Right for VkScript's single-output model this works.

For future we'll need another model that takes into account lots of outputs, in an unstructured way.
What comes to mind:


- Modify DoBuffer to take the buffer name as an argument.
- DoProcessWork : runs queued work, and gets back data for all buffers we've declared
- After DoProcessWork we can call GetBufferData(const std::string& buffer_name, const void** buf); 
      //// and whatever else is needed to be able to scan it out.  But I think the executor has the format and stride info?
    - An empty string for the buffer name will refer to a buffer the engine created for the framebufer because there was no explicitly declared buffer.   (Actually, maybe the executor takes responsibility for creating the implied buffer for the framebuffer?)